src/dotty/tools/dotc/parsing/Scanners.scala

```diff
@@ -182,6 +182,38 @@ object Scanners {
      */
     var revComments: List[Comment] = Nil
 
+    /** The currently closest docstring, replaced every time a new docstring is
+     *  encountered
+     */
+    var closestDocString: List[List[Comment]] = List(List())
```

Comment:
you seem to operate with this list as if it was a Stack. Maybe make it be a stack?